datafusion/core/src/physical_plan/file_format/row_filter.rs

```diff
@@ -98,14 +108,22 @@ impl ArrowPredicate for DatafusionArrowPredicate {
     }
 
     fn evaluate(&mut self, batch: RecordBatch) -> ArrowResult<BooleanArray> {
+        // scoped timer updates on drop
+        let mut timer = self.time.timer();
         match self
             .physical_expr
             .evaluate(&batch)
             .map(|v| v.into_array(batch.num_rows()))
         {
             Ok(array) => {
                 if let Some(mask) = array.as_any().downcast_ref::<BooleanArray>() {
-                    Ok(BooleanArray::from(mask.data().clone()))
+                    let bool_arr = BooleanArray::from(mask.data().clone());
+                    // TODO is there a more efficient way to count the rows that are filtered?
```

Comment:
@tustvold  do you have any suggestions on how to count the number of true values in a boolean array faster/better than this?

Reply:
If you don't care about nulls

```
bool_arr.values().count_set_bits_offset(self.offset(), self.len())
```

If you do care about nulls it is slightly more complicated, I'll get something into arrow-rs

Reply:
https://github.com/apache/arrow-rs/pull/2957

Could copy paste for now, it isn't hugely complicated

Reply:
I do care about nulls, sadly, -- it needs to be Non-null and true.

Reply:
I can file a ticket in arrow-rs if that would be helpful

Reply:
Thanks again @tustvold  -- I filed https://github.com/apache/arrow-rs/issues/2963  and will copy/paste your implementation here for nwo